### ISSUE TITLE
How to refer to SM itself in transition table definition?

### ISSUE THREAD

**Issue by rhaschke** [boost-ext/sml#559]
## Expected Behavior
I want to realize a nested statemachine like below, i.e. the top-level SM should handle events 1,2,3 and transition into the corresponding state. How can I refer to the SM itself in the transition table? Is there any special _self_ state?
![image](https://user-images.githubusercontent.com/5376030/220894662-437802cd-dfcb-4fb4-bfcc-29dec6568acc.png)

## Workaround
I could define a dummy wrapper around SM, which then defines the state transitions for SM on that outer level. However, besides being ugly, this solution makes it impossible to define SM in a self-coherent fashion (for reusability).

**Comment by uyha** [comment]
I'm not sure what you're trying to achieve here, shouldn't something like [this](https://godbolt.org/z/9j3oKsWGG) be enough?

**Comment by rhaschke** [comment]
Yes, thanks a lot: the `state<_>` syntax was exactly what I was looking for.

**Comment by rhaschke** [comment]
@uyha, one more question along that line: Is it possible to create those child states and their corresponding transitions dynamically, i.e. at runtime? As far as I understand, states are identified by type and thus each child would need to have a different type, right? So, runtime composition of a state machine is not possible with SML?

**Comment by uyha** [comment]
I don’t know how to but my guess is it is not possible since SML relies on extremely advanced template meta programming to doing all it does at compile time.

**Comment by rhaschke** [comment]
Yes, that was my impression too. The number of my child states is still manageable to configure the SM manually at compile time. I was just curious.

**Comment by rhaschke** [comment]
@uyha, unfortunately, the proposed solution doesn't work: while the `state<_>` syntax compiles, the state machine doesn't react to any of the declared events this way: https://godbolt.org/z/394qvhc6E.

**Comment by rhaschke** [comment]
> @uyha, unfortunately, the proposed solution doesn't work: while the `state<_>` syntax compiles, the state machine doesn't react to any of the declared events this way: https://godbolt.org/z/jx6azzorT.
